packages/swap/src/quoting/schemas.ts

```diff
@@ -1,45 +1,32 @@
 import { z } from 'zod';
+import { AssetAndChain } from '@/shared/enums';
 import { numericString, unsignedInteger } from '@/shared/parsers';
 import { QuoteType } from './quotes';
 
-export const marketMakerResponseSchema = z
-  .union([
-    z
-      .object({
-        id: z.string(),
-        intermediate_amount: numericString,
-        output_amount: numericString,
-      })
-      .transform(({ id, ...rest }) => ({
-        id,
-        intermediateAmount: rest.intermediate_amount,
-        outputAmount: rest.output_amount,
-      })),
-    z
-      .object({
-        id: z.string(),
-        output_amount: numericString,
-      })
-      .transform(({ id, ...rest }) => ({
-        id,
-        outputAmount: rest.output_amount,
-      })),
-  ])
-  .transform((quote) => ({
-    ...quote,
-    quoteType: 'market_maker' as QuoteType,
-  }));
-export type MarketMakerQuote = z.infer<typeof marketMakerResponseSchema>;
+export const marketMakerResponseSchema = z.object({
+  request_id: z.string(),
+  limit_orders: z.array(z.tuple([z.number(), numericString.transform((n) => BigInt(n))])),
+});
+
+export type MarketMakerRawQuote = z.input<typeof marketMakerResponseSchema>;
+export type MarketMakerQuote = z.output<typeof marketMakerResponseSchema>;
+
+export type MarketMakerQuoteRequest = {
```

Comment:
should we rename this to `JitQuoteRequest`? would be a bit less verbose 🙂

Reply:
It only gets referenced 5 times, so I think it's better to be descriptive